src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -64,37 +68,89 @@ def log_event(event_name, **kwargs):
         elif ARGCOMPLETE_ENV_NAME in os.environ:
             source = 'completer'
 
-        props = {
-            'time': str(time.time()),
-            'x-ms-client-request-id': APPLICATION.session['headers']['x-ms-client-request-id'],
-            'command': APPLICATION.session.get('command', None),
-            'version': core_version,
-            'source': source
-            }
+        types = ['event', 'pageview']
+        if log_type not in types:
+            raise ValueError('Type {} is not supported.  Available types: {}'.format(log_type,
+                                                                                     types))
+
+        profile = Profile()
+        props = {}
+        _safe_exec(props, 'time', lambda: str(datetime.datetime.now()))
+        _safe_exec(props, 'x-ms-client-request-id',
+                   lambda: APPLICATION.session['headers']['x-ms-client-request-id'])
+        _safe_exec(props, 'command', lambda: APPLICATION.session.get('command', None))
+        _safe_exec(props, 'version', lambda: core_version)
+        _safe_exec(props, 'source', lambda: source)
+        _safe_exec(props, 'installation-id', profile.get_installation_id)
+        _safe_exec(props, 'python-version', lambda: _make_safe(str(platform.python_version())))
+        _safe_exec(props, 'shell-type', _get_shell_type)
+        _safe_exec(props, 'locale', lambda: '{},{}'.format(locale.getdefaultlocale()[0],
+                                                           locale.getdefaultlocale()[1]))
+        _safe_exec(props, 'user-machine-id', _get_user_machine_id)
+        _safe_exec(props, 'user-azure-id', _get_user_azure_id)
+        _safe_exec(props, 'azure-subscription-id', _get_azure_subscription_id)
 
         if kwargs:
             props.update(**kwargs)
 
         telemetry_records.append({
-            'name': event_name,
+            'name': name,
+            'type': log_type,
             'properties': props
             })
     except Exception as ex: #pylint: disable=broad-except
         # Never fail the command because of telemetry, unless debugging
         if _debugging():
             raise ex
 
-def _sanitize_inputs(kwargs):
-    for key, value in kwargs.items():
+def _safe_exec(props, key, fn):
+    try:
+        props[key] = fn()
+    except Exception as ex: #pylint: disable=broad-except
+        # Never fail the command because of telemetry, unless debugging
+        if _debugging():
+            raise ex
+
+def _get_user_machine_id():
+    return hash(platform.node() + getpass.getuser())
+
+def _get_user_azure_id():
+    try:
```

Comment:
If there is no active subscription, we throw a CLIError, so we want to ignore those.  If anything else goes wrong, we want to catch it in CI

---
In reply to: [79207141](https://github.com/Azure/azure-cli/pull/910#discussion_r79207141) [](ancestors = 79207141)